mk/spksrc.service.mk

```diff
@@ -239,24 +248,39 @@ ifneq ($(strip $(GROUP)),)
 # or use the shared folder resource worker to add permissions, ask user from wizard see transmission package for an example
 	@jq --arg packagename $(GROUP) '."join-pkg-groupnames" += [{$$packagename}]' $@ 1<>$@
 endif
-endif
 ifneq ($(strip $(SYSTEM_GROUP)),)
 # options: http, system
 	@jq '."join-groupname" = "$(SYSTEM_GROUP)"' $@ 1<>$@
 endif
 ifneq ($(strip $(SPK_USER)),)
 	@jq '."username" = "sc-$(SPK_USER)"' $@ 1<>$@
-ifeq ($(call version_ge, ${TCVERSION}, 7.0),1)
 	@jq '."groupname" = "sc-$(SPK_USER)"' $@ 1<>$@
 endif
+ifneq ($(strip $(SPK_GROUP)),)
+	@jq '."groupname" = "$(SPK_GROUP)"' $@ 1<>$@
+endif
+
+# Less then DSM 7
+else
+ifneq ($(strip $(SYSTEM_GROUP)),)
+# options: http, system
+	@jq '."join-groupname" = "$(SYSTEM_GROUP)"' $@ 1<>$@
+endif
+ifneq ($(strip $(SPK_USER)),)
+	@jq '."username" = "sc-$(SPK_USER)"' $@ 1<>$@
 endif
 ifneq ($(strip $(SPK_GROUP)),)
 	@jq '."groupname" = "$(SPK_GROUP)"' $@ 1<>$@
 endif
+endif
 
+ifneq ("$(wildcard $(DSM_CONF_DIR)/privilege)","")
```

Comment:
This line courses the `conf` folder to be missed inside the spk

Reply:
Would `ifneq (,$(wildcard $(DSM_CONF_DIR)/privilege))` still work for you?

Reply:
I believe it would.
On the other hand, the `wildcard` may need to include `$(DSM_CONF_DIR)/*` or just `$(DSM_CONF_DIR)` so it checks if the directory exists or if there are other files in there and therefore add the `conf` directory if it is the case.  Testing only on the `privilege` file omits the existence of other files.  But enforcing to include the `conf` leads into build failures when it doesn't exists.

Reply:
@publicarray I've changed this to only look for the `conf` directory, and if exists, set `SPK_CONTENT += conf`.
Let me know if that's ok with you.
From the tests I did on my end all looks ok but you might find another corner-case...

Reply:
Sorry but that did not work:

```sh
spksrc/packages/adminer_noarch-dsm7_4.7.6-3 on  dsm6-privilege-fix [📦🤷‍] 
❯ ls -l
.rw-r--r-- 1.2k seb 12 Apr  0:00 INFO
.rw-r--r-- 362k seb 12 Apr  0:00 package.tgz
.rw-r--r-- 4.0k seb 12 Apr  0:00 PACKAGE_ICON.PNG
.rw-r--r--  19k seb 12 Apr  0:00 PACKAGE_ICON_256.PNG
drwxr-xr-x    - seb 12 Apr  0:00 scripts
drwxr-xr-x    - seb 12 Apr  0:00 WIZARD_UIFILES

```

Reply:
@publicarray You are absolutely right,  output led me to think all was good but it wasn't.
I ended-up reworking the code to group per use-case which led to removing `wildcard` call entirely.
Code is now much clearer than I had expected.
Let me know if that looks ok to you.